src/cockpit/channels/filesystem.py

```diff
@@ -179,36 +178,41 @@ async def run(self, options: JsonObject) -> JsonObject:
                     os.unlink(path)
             else:
                 # otherwise, spool data into a temporary file until EOF then rename into place...
-                with tempfile.NamedTemporaryFile(dir=dirname, prefix=f'.{basename}-', delete=False) as tmp:
-                    delete_on_exit = tmp.name
+                fd, tmpname = tempfile.mkstemp(dir=dirname, prefix=f'.{basename}-')
+                try:
                     while data is not None:
-                        await self.in_thread(tmp.write, data)
+                        await self.in_thread(os.write, fd, data)
                         data = await self.read()
 
-                    await self.in_thread(os.fdatasync, tmp.fileno())
+                    await self.in_thread(os.fdatasync, fd)
 
                     if tag is None:
                         # no preconditions about what currently exists or not
                         # calculate the file mode from the umask
-                        os.fchmod(tmp.fileno(), 0o666 & ~my_umask())
-                        os.rename(tmp.name, path)
-                        delete_on_exit = None
+                        os.fchmod(fd, 0o666 & ~my_umask())
+                        os.rename(tmpname, path)
+                        tmpname = ''
```

Comment:
I must say, while this simplifies the typing, something in me prefered `None` for unset values. This feels error prone in situations where an empty string would be wrong, but not wrong enough to trigger a clear exception. In fact, making it `str | None` is even preferable, as it points out places in the code where you forget to check the validity.

Reply:
FTR. This wasn't a hard veto, just a "sigh". If it's too much effort, keep it.

Reply:
ah, you fixed it already, thanks!